crates/relayer/src/chain/ckb4ibc/monitor.rs

```diff
@@ -0,0 +1,357 @@
+                            &String::from_utf8(self.config.client_id.to_vec()).unwrap(),
+                        )
+                        .unwrap(),
+                        counterparty_connection_id: None,
+                        counterparty_client_id: ClientId::from_str(
+                            connection_end.counterparty.client_id.to_str().unwrap(),
+                        )
+                        .unwrap(),
+                    };
+                    let event = IbcEvent::OpenTryConnection(ConnectionOpenTry(attrs));
+                    Some(IbcEventWithHeight {
+                        event,
+                        height: Height::new(1, 1).unwrap(),
+                        tx_hash: tx_hash.clone().into(),
+                    })
+                }
+                _ => None,
+            })
+            .flat_map(|e| e)
+            .collect::<Vec<_>>();
+        Ok(EventBatch {
+            chain_id: self.config.id.clone(),
+            tracking_id: TrackingId::Static("ckb connection events collection"),
+            height: Height::new(1, 1).unwrap(), // todo
+            events,
+        })
+    }
+
+    async fn fetch_channel_events(&self) -> Result<EventBatch> {
+        let key = SearchKey {
+            script: Script::new_builder()
+                .code_hash(self.config.channel_code_hash.pack())
+                .args(
+                    ChannelArgs {
+                        client_id: self.config.client_id.clone(),
+                        open: false,
+                    }
+                    .to_args()
+                    .pack(),
+                )
+                .build()
+                .into(),
+            script_type: ScriptType::Lock,
+            filter: None,
+            with_data: None,
+            group_by_transaction: None,
+        };
+        let identified_channel_ends = self
+            .search_and_extract(
+                key,
+                &|tx| {
+                    let hash = tx.hash.clone();
+                    let obj = extract_channel_end_from_tx(tx)
+                        .map_err(|_| Error::collect_events_failed("channel".to_string()))?;
+                    Ok((obj, hash))
+                },
+                20,
+            )
+            .await?;
+
+        let events = identified_channel_ends
+            .into_iter()
+            .map(|item| match item.0.channel_end.state {
+                State::Init => IbcEventWithHeight {
+                    event: IbcEvent::OpenInitChannel(ChannelOpenInit {
+                        port_id: item.0.port_id,
+                        channel_id: Some(item.0.channel_id),
+                        connection_id: item.0.channel_end.connection_hops[0].clone(),
+                        counterparty_port_id: item.0.channel_end.remote.port_id,
+                        counterparty_channel_id: item.0.channel_end.remote.channel_id,
+                    }),
+                    height: Height::new(1, 1).unwrap(), // todo
```

Comment:
I suggest using `u64::MAX` instead of `1` for the `Height` object to represent a nonsense case